### ISSUE TITLE
Fix S3060 FN: Support list patterns

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6295]
### Description

FN for S3060 (DoNotTestThisWithIsOperator).

Using C#11 list patterns, the rule is not raising an issue.

### Repro steps

```csharp
   var check = this is [1, 2, 3]; // FN
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.
